### PR TITLE
F/aws fsx windows file system: Support Support updating throughput_capacity and Storage_Capacity

### LINKED ISSUE #15558: r/aws_fsx_windows_file_system: Support updating throughput_capacity
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

It's not currently possible to update the throughput for a file system using Terraform - the `throughput_capacity` attribute has `ForceNew` configured in the schema. Newer filesystems can be updated.

### New or Affected Resource(s)

* `aws_fsx_windows_file_system`

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* https://docs.aws.amazon.com/fsx/latest/WindowsGuide/managing-throughput-capacity.html#increase-throughput-capacity

### PR DESCRIPTION
**Pull request by nikhil-goenka**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #15558

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note

```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSFsxWindowsFileSystem_ThroughputCapacity'
--- PASS: TestAccAWSFsxWindowsFileSystem_ThroughputCapacity (3546.05s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	3548.249s
$ make testacc TESTARGS='-run=TestAccAWSFsxWindowsFileSystem_StorageCapacity'
--- PASS: TestAccAWSFsxWindowsFileSystem_StorageCapacity (2681.58s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	2683.927s
...
```

### CODE REVIEW

**Review by DrFaust92 — APPROVED** [review]
LGTM

```
--- PASS: TestAccAWSFsxWindowsFileSystem_StorageCapacity (3376.82s)
--- PASS: TestAccAWSFsxWindowsFileSystem_basic (3243.67s)
--- PASS: TestAccAWSFsxWindowsFileSystem_ThroughputCapacity (3338.47s)
```

**Comment by breathingdust** [comment]
LGTM :rocket: Thanks @nikhil-goenka for the contribution!

Verified Acceptance Tests in Commercial (us-west-2)

```console
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSFsxWindowsFileSystem_ThroughputCapacity'

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSFsxWindowsFileSystem_ThroughputCapacity -timeout 120m
=== RUN   TestAccAWSFsxWindowsFileSystem_ThroughputCapacity
=== PAUSE TestAccAWSFsxWindowsFileSystem_ThroughputCapacity
=== CONT  TestAccAWSFsxWindowsFileSystem_ThroughputCapacity
--- PASS: TestAccAWSFsxWindowsFileSystem_ThroughputCapacity (3631.91s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	3633.552s

make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSFsxWindowsFileSystem_StorageCapacity'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSFsxWindowsFileSystem_StorageCapacity -timeout 120m
=== RUN   TestAccAWSFsxWindowsFileSystem_StorageCapacity
=== PAUSE TestAccAWSFsxWindowsFileSystem_StorageCapacity
=== CONT  TestAccAWSFsxWindowsFileSystem_StorageCapacity
--- PASS: TestAccAWSFsxWindowsFileSystem_StorageCapacity (3261.52s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	3263.013s

make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSFsxWindowsFileSystem_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSFsxWindowsFileSystem_basic -timeout 120m
=== RUN   TestAccAWSFsxWindowsFileSystem_basic
=== PAUSE TestAccAWSFsxWindowsFileSystem_basic
=== CONT  TestAccAWSFsxWindowsFileSystem_basic
--- PASS: TestAccAWSFsxWindowsFileSystem_basic (3277.35s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	3278.789s
```

File: aws/fsx.go [hunk 1]
```diff
@@ -48,6 +48,26 @@ func refreshFsxFileSystemLifecycle(conn *fsx.FSx, id string) resource.StateRefre
...
+
+		if err != nil {
+			return nil, "", err
+		}
+
+		if filesystem == nil {
+			return nil, "", nil
+		}
+
+		return filesystem, aws.StringValue(filesystem.AdministrativeActions[0].Status), nil
```
Comment (bflad) [inline aws/fsx.go:67]: This line is missing a length check on `AdministrativeActions`, which can cause a panic. Created followup issue: https://github.com/hashicorp/terraform-provider-aws/issues/16440

File: aws/fsx.go [hunk 2]
```diff
@@ -89,3 +109,20 @@ func waitForFsxFileSystemUpdate(conn *fsx.FSx, id string, timeout time.Duration)
 
 	return err
 }
+
+func waitForFsxFileSystemUpdateOptimizing(conn *fsx.FSx, id string, timeout time.Duration) error {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{fsx.StatusInProgress},
```
Comment (bflad) [inline aws/fsx.go:115]: Seems like FSx also uses a `PENDING` status before it goes in progress. Created followup issue: https://github.com/hashicorp/terraform-provider-aws/issues/16440

**Pull request merged**